src/sys/ptrace/linux.rs

```diff
@@ -233,6 +268,33 @@ fn ptrace_get_data<T>(request: Request, pid: Pid) -> Result<T> {
     Ok(unsafe{ data.assume_init() })
 }
 
+/// Function for ptrace requests that return values from the data field.
+/// Some ptrace get requests populate structs or larger elements than `c_long`
+/// and therefore use the data field to return values. This function handles these
+/// requests.
+fn ptrace_getregset<T>(request: Request, pid: Pid, regset: i32) -> Result<T> {
+    let mut data = mem::MaybeUninit::<T>::uninit();
+    let expected_len = mem::size_of::<T>();
+    let mut output = iovec {
+        iov_base: data.as_mut_ptr() as *mut _ as *mut c_void,
+        iov_len: expected_len,
+    };
+
+    let res = unsafe {
+        libc::ptrace(request as RequestType,
+                     libc::pid_t::from(pid),
+                     regset as usize as *const c_void,
+                     &mut output as *mut _ as *mut c_void)
+    };
+    // If we get the struct definition wrong for a platform, linux can just scribble all over
+    // our memory. In some sense it's "too late" because it could have written past the bounds
+    // of our `data` value, but it's better to try to crash now than keep the program going!
```

Comment:
I don't think this comment is accurate. From what I understand reading the ptrace docs, it won't write past the end of our buffer since it knows that from the input `iov_len`. It sets the length of `iov_len` so that we can know if our buffer is too small or too large for the full set of registers. So, there shouldn't be any stack corruption here as far as I can tell (but this assertion is still great to have).